src/components/VideoPlayerContexts/FullScreenContext.tsx

```diff
@@ -0,0 +1,34 @@
+import React, {useCallback, useContext, useMemo, useRef} from 'react';
+import type WindowDimensions from '@hooks/useWindowDimensions/types';
+import type ChildrenProps from '@src/types/utils/ChildrenProps';
+import type {FullScreenContext} from './types';
+
+const Context = React.createContext<FullScreenContext | null>(null);
+
+function FullScreenContextProvider({children}: ChildrenProps) {
+    const isFullscreen = useRef(false);
+    const lockedWindowDimensions = useRef<WindowDimensions | null>(null);
+
+    const lockWindowDimensions = useCallback((newWindowDimensions: WindowDimensions) => {
+        lockedWindowDimensions.current = newWindowDimensions;
+    }, []);
+
+    const unlockWindowDimensions = useCallback(() => {
+        lockedWindowDimensions.current = null;
+    }, []);
+
+    const contextValue = useMemo(() => ({isFullscreen, lockedWindowDimensions, lockWindowDimensions, unlockWindowDimensions}), [unlockWindowDimensions, lockWindowDimensions]);
+    return <Context.Provider value={contextValue}>{children}</Context.Provider>;
+}
+
+function useFullScreenContext() {
+    const context = useContext(Context);
+    if (context === undefined) {
+        throw new Error('usePlaybackContext must be used within a FullScreenContextProvider');
+    }
+    return context;
+}
+
+FullScreenContextProvider.displayName = 'FullScreenContextProvider';
+
+export {Context as FullScreenContext, FullScreenContextProvider, useFullScreenContext};
```

Comment:
why not use the name `FullScreenContext` in line 6 instead of generic Context and then don't use the `as` here.

Reply:
Because we have a context type that is named like that. I was thinking about changing that in the way you suggest, however, I saw that in other places (like `PlaybackContext` or `VolumeContext`) it is done the same, the name is taken, so it wanted to follow the schema

Reply:
alright, Im not the biggest fan of this naming style (because in the wider context we have ex `PlaybackContext ` that could mean 2 things) but keeping this consistent with the rest is a good idea 👍